rlkit/samplers/in_place.py

```diff
@@ -40,7 +40,7 @@ def obtain_samples(self, deterministic=False, max_samples=np.inf, max_trajs=np.i
         n_trajs = 0
         while n_steps_total < max_samples and n_trajs < max_trajs:
             path = rollout(
-                self.env, policy, max_path_length=self.max_path_length)
+                self.env, policy, max_path_length=self.max_path_length, update_context=update_context)
```

Comment:
hmm does the context update need to happen inside `rollout()`? I was imagining we could just collect the trajectories as paths, and then add them to the context as a batch whenever we want to update the posterior.

Reply:
I kept it there as an option so we could update at transition level.

Reply:
I guess a more clean way of handling this would be to separate update context and update posterior.
Update context would just append a data point to the agents context buffer.
Update posterior would compute new distribution from current context (and probably sample a new z).
And then we would only need to explicitly resample z if we wanted keep the randomness from the standard gaussian sample fixed while updating the parameters with which we transform, which I'm not sure we really need here.